x-pack/plugins/security_solution/public/entity_analytics/components/risk_score_preview_section.tsx

```diff
@@ -0,0 +1,191 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import {
+  EuiAccordion,
+  EuiFormRow,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiCallOut,
+  EuiButton,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+import type { BoolQuery, Query } from '@kbn/es-query';
+import { buildEsQuery } from '@kbn/es-query';
+import { RiskScoreEntity } from '../../../common/risk_engine/types';
+import { RiskScorePreviewTable } from './risk_score_preview_table';
+import * as i18n from '../translations';
+import type { RiskScore } from '../../../server/lib/risk_engine/types';
+import { useRiskScorePreview } from '../api/hooks/use_preview_risk_scores';
+import { useKibana } from '../../common/lib/kibana';
+import { SourcererScopeName } from '../../common/store/sourcerer/model';
+import { useSourcererDataView } from '../../common/containers/sourcerer';
+
+interface IRiskScorePreviewPanel {
+  showMessage: string;
+  hideMessage: string;
+  isLoading: boolean;
+  items: RiskScore[];
+  type: RiskScoreEntity;
+}
+
+const getRiskiestScores = (scores: RiskScore[] = [], field: string) =>
+  scores
+    ?.filter((item) => item?.identifierField === field)
+    ?.sort((a, b) => b?.totalScoreNormalized - a?.totalScoreNormalized)
+    ?.slice(0, 5) || [];
+
+const RiskScorePreviewPanel = ({
+  items,
+  showMessage,
+  hideMessage,
+  isLoading,
+  type,
+}: IRiskScorePreviewPanel) => {
+  const [trigger, setTrigger] = useState<'closed' | 'open'>('open');
+  const onToggle = (isOpen: boolean) => {
+    const newState = isOpen ? 'open' : 'closed';
+    setTrigger(newState);
+  };
+
+  return (
+    <EuiPanel hasBorder={true}>
+      <EuiAccordion
+        initialIsOpen={true}
+        isLoading={isLoading}
+        id={'host-table'}
+        buttonContent={trigger === 'closed' ? showMessage : hideMessage}
+        forceState={trigger}
+        onToggle={onToggle}
+        extraAction={<EuiIcon type={type === RiskScoreEntity.host ? 'storage' : 'user'} />}
+      >
+        <>
+          <EuiSpacer size={'m'} />
+          <RiskScorePreviewTable items={items} type={type} />
+        </>
+      </EuiAccordion>
+    </EuiPanel>
+  );
+};
+
+export const RiskScorePreviewSection = () => {
+  const [query, setQuery] = useState<{
+    dateRange: { from: string; to: string };
+    query: Query;
+  }>({
+    dateRange: { from: 'now-24h', to: 'now' },
+    query: { query: '', language: 'kuery' },
+  });
+
+  const {
+    unifiedSearch: {
+      ui: { SearchBar },
+    },
+  } = useKibana().services;
+
+  let riskScorePreviewFilter: { bool: BoolQuery } = {
+    bool: { must: [], filter: [], should: [], must_not: [] },
+  };
+  try {
+    riskScorePreviewFilter = buildEsQuery(undefined, query.query, []);
+  } catch (e) {
+    // Default to keeping default filter
```

Comment:
Should we display an error to the user in this case?

Reply:
bump

Reply:
Yes, I added a toast error, as we do in searchbar in other pages

Reply:
Sorry, I didn't see that this one was outdated.